src/Raven.Client/Documents/Indexes/AbstractIndexCreationTask.cs

```diff
@@ -223,6 +228,12 @@ public virtual Task ExecuteAsync(IDocumentStore store, DocumentConventions conve
                 if (State.HasValue)
                     indexDefinition.State = State.Value;
 
+                if (SourceItemKind.HasValue)
+                {
+                    if (indexDefinition.SourceType != IndexSourceType.Documents)
+                        throw new NotSupportedException($"{nameof(SourceItemKind)} can be set only for document indexes.");
```

Comment:
We need to have this validation on server side as well. People can put an index definition directly using HTTP call or from other languages

Reply:
noted, I will find good place for this validation